### PR TITLE
Tracing improvements

### PR DESCRIPTION
**Pull request by coot**
- mini-protocols: trace termination message
- tx-submission-inbound: improved logging

### CODE REVIEW

File: ouroboros-network/protocol-tests/Ouroboros/Network/Protocol/TxSubmission/Direct.hs [hunk 1]
```diff
@@ -39,8 +40,7 @@ directPipelined (TxSubmissionServerPipelined mserver)
         SendMsgReplyTxIds (BlockingReply txids) client' -> do
           server' <- serverNext txids
           directSender q server' client'
-        SendMsgDone b ->
-          return (a, b)
+        SendMsgDone b -> (,b) <$> a
```
Comment (karknu) [inline ouroboros-network/protocol-tests/Ouroboros/Network/Protocol/TxSubmission/Direct.hs:43]: Is this change really needed?

**Review by karknu — APPROVED** [review]
LGTM.

File: ouroboros-network/protocol-tests/Ouroboros/Network/Protocol/TxSubmission/Direct.hs (hunk 1 above)
Comment (coot) [reply ouroboros-network/protocol-tests/Ouroboros/Network/Protocol/TxSubmission/Direct.hs:43]: Yes, because `a :: m a`, while it used to be `a :: a`; using monadic action allows to log a message when sending `MsgDone`.

**Comment by coot** [comment]
bors merge
